crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -827,14 +827,19 @@ impl<T: TransactionOrdering> TxPool<T> {
         tx_hash: &B256,
     ) -> Option<Arc<ValidPoolTransaction<T::Transaction>>> {
         let (tx, pool) = self.all_transactions.remove_transaction_by_hash(tx_hash)?;
+
+        // After tx is removed, its descendants must become parked due to the nonce gap
+        let updates = self.all_transactions.park_descendant_pending_txs(tx.id());
```

Comment:
This change makes sense for mined txs but does it also need to consider whether this path could be called for re-org scenarios? If a mined tx is reverted due to a chain re-org, would its descendants be requeued correctly, or could it end up with inconsistent subpool state?